sklearn/ensemble/_bagging.py

```diff
@@ -120,10 +129,23 @@ def _parallel_build_estimators(
                 not_indices_mask = ~indices_to_mask(indices, n_samples)
                 curr_sample_weight[not_indices_mask] = 0
 
-            estimator.fit(X[:, features], y, sample_weight=curr_sample_weight)
+            if has_check_input:
```

Comment:
I think we can use [functools.partial](https://docs.python.org/3/library/functools.html#functools.partial) so we do not need to branch this much. Specifically before `if support_sample_weight`:

```python
if has_check_input:
    estimator_fit = partial(estimator.fit, check_input=check_input)
else:
    estimator_fit = estimator.fit
```

and then use `estimator_fit` to fit the estimator here.